### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by eamador1**
Since the last code review, the following was done in this project:
___

- Create a class Classroom ✅
- Create the has-many/belongs-to relationship between Classroom and Student ✅
- Create a class Book ✅
- Create a class Rental ✅
- Create the many-to-many (also has-many-through) relationship between Person and Book ✅

### CODE REVIEW

File: book.rb
Comment (fobadara) [inline book.rb:1]: [REQUIRED] You should have a `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. So your book should have many rentals like this:
```ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb
Comment (fobadara) [inline person.rb:1]: [REQUIRED] You should have a many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. So your person should have many rentals like this:
```ruby
 def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @eamador1, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

REDO

## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 1]
```diff
@@ -25,4 +26,8 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(book, date)
+    Rental.new(date, book, self)
+  end
```
Comment (zeff96) [inline person.rb]: - [ ] Hey there! 👋 Fantastic job on the implementation of a method that handles many-to-many relationships through rentals. However, it's crucial to highlight that the current setup designates this method as private. To enhance clarity and maintain best practices, let's relocate it from the private methods section. 🚀 This adjustment ensures better visibility and accessibility to this important functionality. Keep up the awesome work! 👏

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️
Hi @eamador1 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost

 ## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Descriptive PR


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (rloterh) [reply person.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Erwin,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on introducing proper encapsulation!

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**